File: fornt-location/src/app/models/car.ts
export interface Car {
  id: number;
  city: string; // Par exemple, le nom de la ville ou un identifiant
  type: string; // Par exemple, le type de la voiture
  fuel: string; // Par exemple, "Essence", "Diesel", etc.
  gearbox: string; // Par exemple, "Manuelle", "Automatique"
  number_place: number;
  number_door: number;
  clim: boolean;
  number_bag: number;
  caution: number;
  url_photo: string;
  age_min: number;
  driving_license_min_year: number;
  min_age:number;
  model: string;
  brand: string; // Par exemple, le nom de la marque
  photoFile: File | null;
}

File: location_des_voitures-main/location_voiture/src/main/java/com/example/location_voiture/entities/Car.java
package com.example.location_voiture.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Table(name = "car")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Car {



    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private City city;

    @Enumerated(EnumType.STRING)
    private Type type;

    @Enumerated(EnumType.STRING)
    private Fuel fuel;

    @Enumerated(EnumType.STRING)
    private Gearbox gearbox;

    @Column(nullable = false)
    private int number_place;

    @Column(nullable = false)
    private int number_door;

    @Column(nullable = false)
    private boolean clim;

    @Column(nullable = false)
    private int number_bag;

    @Column(nullable = false)
    private float caution;

    @Column(nullable = false)
    private String url_photo;

    @Column(nullable = false)
    private int age_min;

    @Column(nullable = false)
    private int driving_license_min_year;

    @Column(nullable = false)
    private String model;

    @Enumerated(EnumType.STRING)
    private Brand brand;


    @ManyToOne
    @JoinColumn(name = "agency_id", referencedColumnName = "id")
    private Agency agency;

}